api/src/services/payload.ts

```diff
@@ -261,6 +261,15 @@ export class PayloadService {
 	processDates(payloads: Partial<Record<string, any>>[], action: Action): Partial<Record<string, any>>[] {
 		const fieldsInCollection = Object.entries(this.schema.collections[this.collection].fields);
 
+		// Update specific database type overrides
+		for (const [_name, field] of fieldsInCollection) {
+			if (field.special.includes('sqlite-timestamp-in-datetime')) {
```

Comment:
Lets drop the database specific names from these types. We could just rename those flags from `sqlite-timestamp-in-datetime` to `cast-timestamp` and `cast-datetime` respectively. We should do the same thing for the other casting types (`cast-boolean`, `cast-json` etc)

Reply:
@rijkvanzanten Done for `cast-timestamp` and `cast-datetime`.
Changing the existing flags will require an update to the DB itself.
So should the following exhaustive list be updated?
`uuid` => `cast-uuid`
`boolean` => `cast-boolean`
`json` => `cast-json`
`csv` => `cast-csv`

https://github.com/directus/directus/blob/59a72984bfb275204b0bb69368c9e24689fb5a88/api/src/services/payload.ts#L59-L134

Reply:
> So should the following exhaustive list be updated?

Yes, but lets do that in a different PR / cycle 🙂

Reply:
@rijkvanzanten  Oh dear, I've added the following into this PR, should I undo and extract to another PR?

> `uuid` => `cast-uuid`
> `boolean` => `cast-boolean`
> `json` => `cast-json`
> `csv` => `cast-csv`
> `hash` => `cast-hash`

Added the `hash` type into the list too.
`geometry` is left untouched as it is using field types like `geometry.Point` and `geometry.LineString`.